wallets/service.go

```diff
@@ -0,0 +1,176 @@
+}
+
+// Count returns the number of claimed or unclaimed deposit addresses.
+func (hd *HD) GetCountClaimedDepositAddresses(ctx context.Context, claimed bool) (int, error) {
+	c, err := hd.db.Count(ctx, claimed)
+	return int(c), ErrWalletsService.Wrap(err)
+}
+
+// GetAccount returns the info related to an address.
+func (hd *HD) GetAccount(ctx context.Context, address []byte) (account Account, err error) {
+	a, err := hd.db.Get(ctx, address)
+	return Account{Address: a.Address, Claimed: &a.Claimed}, ErrWalletsService.Wrap(err)
+}
+
+// Setup is used to create wallets for test purposes.
+func (hd *HD) Setup(ctx context.Context, size int) (firstAddr []byte, err error) {
+	return firstAddr, err
+}
+
+//--- Implementation for testing ---//
+
+// HD_test implements Wallet interface. Represents hierarchical deterministic wallets.
+type HD_test struct {
+	log    *zap.Logger
+	db     *storjscandb.WalletsDB
+	wallet *mm.Wallet
+}
+
+// NewHD_test creates a new HD_test struct.
+func NewHD_test(log *zap.Logger, db *storjscandb.WalletsDB) (*HD_test, error) {
+	seed, err := mm.NewSeed()
+	if err != nil {
+		return &HD_test{}, err
+	}
+	w, err := mm.NewFromSeed(seed)
+	if err != nil {
+		return &HD_test{}, err
+	}
+	return &HD_test{
+		log:    log,
+		db:     db,
+		wallet: w,
+	}, nil
+}
+
+// GetNewDepositAddress returns the next unclaimed deposit address and claims it.
+func (hd *HD_test) GetNewDepositAddress(ctx context.Context) (address []byte, err error) {
+	defer mon.Task()(&ctx)(&err)
+	wallet, err := hd.db.GetNextAvailable(ctx)
+	if err != nil {
+		return address, ErrWalletsService.Wrap(err)
+	}
+	_, err = hd.db.Claim(ctx, wallet.Address)
+	return wallet.Address, ErrWalletsService.Wrap(err)
+}
+
+// CountTotal returns the total number of deposit addresses.
+func (hd *HD_test) GetCountDepositAddresses(ctx context.Context) (int, error) {
+	total, err := hd.db.TotalCount(ctx)
+	return int(total), ErrWalletsService.Wrap(err)
+}
+
+// Count returns the number of claimed or unclaimed deposit addresses.
+func (hd *HD_test) GetCountClaimedDepositAddresses(ctx context.Context, claimed bool) (int, error) {
+	c, err := hd.db.Count(ctx, claimed)
+	return int(c), ErrWalletsService.Wrap(err)
+}
+
+// GetAccount returns the info related to an address.
+func (hd *HD_test) GetAccount(ctx context.Context, address []byte) (account Account, err error) {
+	a, err := hd.db.Get(ctx, address)
+	return Account{Address: a.Address, Claimed: &a.Claimed}, ErrWalletsService.Wrap(err)
+}
+
+// Setup is used to create wallets for test purposes.
+// Is there a better way to do this?
+func (hd *HD_test) Setup(ctx context.Context, size int) (firstAddr []byte, err error) {
+	firstAddr, err = hd.generateNewAccounts(ctx, size)
+	return firstAddr, ErrWalletsService.Wrap(err)
+}
+
+//--- helper methods for hd_test. NB: similar functions will be in a command line tool for production. ---/
+func (hd *HD_test) generateNewAccounts(ctx context.Context, size int) (firstAddr []byte, err error) {
```

Comment:
nit: what is the purpose of returning the first address here? This is a nit, as this helper function may be used for testing purposes only, but I think returning the number of records created/updated in batch calls to the database may be more useful, perhaps even in addition to what you have already if there is a case for wanting the first record.

Reply:
Returning the first address here was just used for testing purposes, but yeah we can return the number of records created/updated if that makes more sense.